### ISSUE TITLE
[Feature Request] Port features in GluonCV, GluonNLP..., back to MXNet

### ISSUE THREAD

**Issue by sxjscience** [apache/incubator-mxnet#17289]
## Description
Some useful functionalities of MXNet are implemented in GluonCV, GluonNLP, etc.. I think we should port some reusable features back to MXNet to improve the user's experience. For example, we can port the following extensions

- batchify function: https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/data/batchify/batchify.py
- learning rate scheduler: https://github.com/dmlc/gluon-cv/blob/master/gluoncv/utils/lr_scheduler.py
- initializers: https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/initializer/initializer.py

## Related Issues:
https://github.com/apache/incubator-mxnet/issues/16767, https://github.com/apache/incubator-mxnet/issues/17224
